File: token/Controllers/TokenController.cs
using Microsoft.AspNetCore.Mvc;
using token.Dtos;
using token.Models;
using token.Services;

namespace token.Controllers;

public class TokenController : ControllerBase
{
    private readonly ITokenService _tokenService;

    public TokenController(ITokenService tokenService)
    {
        _tokenService = tokenService;
    }


    public ActionResult<ResponseTokenDto<string>> CreateToken(string userId)
    {
        return _tokenService.CreateToken(userId);
    }

    public ActionResult<ResponseTokenDto<string>> DestroyToken(string token)
    {
      
        return _tokenService.DestroyToken(token);
    }

    public ActionResult<ResponseTokenDto<Token>> DecodeToken(string token)
    {
        return _tokenService.DecodeToken(token);
    }
}

File: token/token/src/controllers/token.controller.ts
import {Controller, Get} from '@nestjs/common';
import {TokenService} from '../services/token.service';
import {MessagePattern} from "@nestjs/microservices";
import {MessagePatternEnum} from "../interfaces/message-enums/message-pattern.enum";
import {ResponseTokenDto} from "../interfaces/response-dtos.ts/response-token.dto";
import {IToken} from "../interfaces/IToken";

@Controller()
export class TokenController {
    constructor(private readonly tokenService: TokenService) {
    }

    @MessagePattern(MessagePatternEnum.TOKEN_CREATE)
    async createToken(userId: string): Promise<ResponseTokenDto<IToken>> {
        return await this.tokenService.createToken(userId);
    }

    @MessagePattern(MessagePatternEnum.TOKEN_DESTROY)
    async destroyToken(userId: string): Promise<ResponseTokenDto<string>> {
        return await this.tokenService.destroyToken(userId);
    }

    @MessagePattern(MessagePatternEnum.TOKEN_DECODE)
    async decodeToken(token: string): Promise<ResponseTokenDto<string>> {
        return await this.tokenService.decodeToken(token);
    }


}
